src/script/microbit-interfacing/MicrobitSerial.ts

```diff
@@ -104,26 +130,54 @@ export class MicrobitSerial implements MicrobitConnection {
 
       stateOnAssigned(DeviceRequestStates.INPUT, this.usb.getModelNumber());
       stateOnReady(DeviceRequestStates.INPUT);
+      logMessage('Serial successfully connected');
     } catch (e) {
       logError('Failed to initialise serial protocol', e);
       stateOnFailedToConnect(DeviceRequestStates.INPUT);
       await this.usb.stopSerial();
       throw e;
```

Comment:
Interesting.. for some reasons the first reconnect after connection loss (clicking reset btn on micro:bit) currently fail. The error message includes: `Error: Timeout waiting for response 530415182`

Reply:
With help from Matt, this seems to be fixed by stopping serial before trying to reconnect